src/main/java/Duke.java

```diff
@@ -1,10 +1,102 @@
+        for (int i = 0; i < taskSum; i++) {
+            System.out.println((i+1) + ". " + "[" + tasks[i].getStatusIcon() + "] " + tasks[i].getName());
+        }
+        System.out.println(divider);
+    }
+
+    public static void markAsDone(Task[]tasks, int taskNum, int taskSum) {
+        String divider = "______________________________";
+        int taskIndex = taskNum - 1;
+        if (taskNum > taskSum) {
+            System.out.println(divider + System.lineSeparator()
+                    + "Unable to mark task as done since the task does not exist!"
+                    + System.lineSeparator() + divider);
+        } else {
+            tasks[taskIndex].isDone = true;
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println("[X] " + tasks[taskIndex].getName()
+                    + System.lineSeparator() + divider);
+        }
+    }
+
+    public static void markAsNotDone(Task[]tasks, int taskNum, int taskSum) {
+        String divider = "______________________________";
+        int taskIndex = taskNum - 1;
+        if (taskNum > taskSum) {
+            System.out.println(divider + System.lineSeparator()
+                    + "Unable to mark task as not done since the task does not exist!"
+                    + System.lineSeparator() + divider);
+        } else {
+            tasks[taskIndex].isDone = false;
+            System.out.println("OK, I've marked this task as not done yet:");
+            System.out.println("[ ] " + tasks[taskIndex].getName()
+                    + System.lineSeparator() + divider);
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String input;
+        //String[] tasks = new String[100];
+        //Use tasks class instead of string class
+        Task[] list = new Task[100];
+        Scanner in = new Scanner(System.in);
+        greetUser();
+        int taskSum = 0;
+        int taskNum;
+        boolean end = false;
+
+        while(!end) {
+            input = in.nextLine();
+            String[] inputText = input.split(" ");
+            if(input.equals("bye")) {
+                end = true;
+            } else if(input.equals("list")) {
+                printList(list, taskSum);
+            } else if (inputText[0].equals("mark")) {
+                taskNum = Integer.parseInt(inputText[1]);
+                markAsDone(list, taskNum, taskSum);
+            } else if(inputText[0].equals("unmark")) {
+                taskNum = Integer.parseInt(inputText[1]);
+                markAsNotDone(list,taskNum, taskSum);
+            } else {
+                echoUserInput(input);
+                addToList(list, input, taskSum);
```

Comment:
The final else should mean 'everything else', and not the final option here.
This will be useful for your level-4 requirements to sort which type of task is being entered.
The else statement can be used to catch errors and helpful for debugging.
https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#use-the-default-branch